plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,633 @@
+EXPORT_NO_RESULTS_MSG = "Unable to export the diagnostics results because the results do not exist."
+SUCCESS_RUN_MSG = "Successfully run the diagnostics operation."
+SUCCESS_RUN_AND_EXPORT_MSG = "Successfully run and exported the diagnostics."
+RUNNING_RUN_MSG = "Successfully triggered the job to run diagnostics."
+INVALID_FILE_MSG = "File extension is invalid. Supported extension is: .txt."
+INVALID_DIRECTORY_MSG = "Provided directory path '{path}' is not valid."
+INSUFFICIENT_DIRECTORY_PERMISSION_MSG = "Provided directory path '{path}' is not writable. " \
+                                        "Please check if the directory has appropriate permissions"
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The parameter `job_wait_timeout` value cannot be negative or zero."
+WAIT_TIMEOUT_MSG = "The job is not complete after {0} seconds."
+START_TIME = "The specified scheduled start time occurs in the past, " \
+             "provide a future time to schedule the start time."
+INVALID_TIME = "The specified date and time `{0}` to schedule the diagnostics is not valid. Enter a valid date and time."
+END_START_TIME = "The end time `{0}` to schedule the diagnostics must be greater than the start time `{1}`."
+
+IGNORE_CERTIFICATE_WARNING = {"off": "Off", "on": "On"}
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class ExportDiagnostics:
+    STATUS_SUCCESS = [200, 202]
+
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        try:
+            share_type = module.params.get('share_parameters').get('share_type')
+            export_diagnostics_url = self.__get_export_diagnostics_url(module)
+            job_status = {}
+            self.__check_file_extension(module)
+            if share_type == "local":
+                export_diagnostics_status = self.__export_diagnostics_local(module, export_diagnostics_url)
+            elif share_type in ["http", "https"]:
+                export_diagnostics_status = self.__export_diagnostics_http(module, export_diagnostics_url)
+                job_status = self.get_job_status(module, export_diagnostics_status)
+            elif share_type == "cifs":
+                export_diagnostics_status = self.__export_diagnostics_cifs(module, export_diagnostics_url)
+                job_status = self.get_job_status(module, export_diagnostics_status)
+            elif share_type == "nfs":
+                export_diagnostics_status = self.__export_diagnostics_nfs(module, export_diagnostics_url)
+                job_status = self.get_job_status(module, export_diagnostics_status)
+            status = export_diagnostics_status.status_code
+            if status in self.STATUS_SUCCESS:
+                msg = SUCCESS_EXPORT_MSG
+                job_details = job_status
+                return msg, job_details
+        except HTTPError as err:
```

Comment:
Better if we can handle it the main method itself instead of here.

Reply:
done